tests/api2/interfaces.py

```diff
@@ -10,29 +10,134 @@
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list) is True, results.text
     netmask = results.json()[0]['state']['aliases'][1]['netmask']
-    assert netmask == int(BRIDGENETMASK), results.text
+    aliases['netmask'] = netmask
+    assert isinstance(netmask, int) is True, results.text
+
+
+def test_06_set_main_interfaces_ipv4_to_false():
+    payload = {'ipv4_dhcp': False}
+    results = PUT(f'/interfaces/id/{interface}/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict) is True, results.text
+    assert results.json()['ipv4_dhcp'] is False, results.text
+
+
+def test_07_looking_main_interfaces_ipv4_dhcp_if_is_false():
+    results = GET(f'/interfaces/id/{interface}')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict) is True, results.text
+    assert results.json()['ipv4_dhcp'] is False, results.text
+
+
+def test_08_creating_vlan1_interface():
+    global payload
+    payload = {
+        "ipv4_dhcp": True,
+        "name": "vlan2",
+        "vlan_parent_interface": interface,
+        "type": "VLAN",
+        "vlan_tag": 1,
+        "vlan_pcp": 1
+    }
+    global results
+    results = POST('/interfaces/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict) is True, results.text
+    global interfaces_id
+    interfaces_id = results.json()['id']
+
+
+@pytest.mark.parametrize('dkey', ["ipv4_dhcp", "name", "vlan_parent_interface",
+                                  "type", "vlan_tag", "vlan_pcp"])
+def test_09_looking_at_vlan1_created_interfaces_results_output_(dkey):
+    assert results.json()[dkey] == payload[dkey], results.text
+
+
+def test_10_get_the_vlan1_interface_results():
+    global results
+    results = GET(f'/interfaces/id/{interfaces_id}/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict) is True, results.text
+
+
+@pytest.mark.parametrize('dkey', ["ipv4_dhcp", "name", "vlan_parent_interface",
+                                  "type", "vlan_tag", "vlan_pcp"])
+def test_11_looking_at_vlan1_interface_results_output_(dkey):
+    assert results.json()[dkey] == payload[dkey], results.text
+
+
+def test_12_get_interfaces_ipv4_in_use():
+    global results
+    results = POST("/interfaces/ip_in_use/", {"ipv4": True})
+    assert results.status_code == 200, results.text
+
+
+@pytest.mark.parametrize('dkey', ['type', 'address', 'netmask', 'broadcast'])
+def test_13_look_at_interfaces_ipv4_in_use_output_(dkey):
+    assert results.json()[0][dkey] == aliases[dkey], results.text
+
+
+def test_14_get_interfaces_checkin_waiting():
+    results = GET('/interfaces/checkin_waiting/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), bool) is True, results.text
+
+
+def test_15_get_interfaces_checkin():
+    results = GET('/interfaces/checkin/')
```

Comment:
You should call /interfaces/commit/ to apply all the interfaces changes made before. You need to call /checkin/ after /commit/